python/ray/tune/ray_trial_executor.py

```diff
@@ -94,7 +94,7 @@ def _setup_remote_runner(self, trial, reuse_allowed):
         if self._cached_actor:
             logger.debug("Cannot reuse cached runner {} for new trial".format(
                 self._cached_actor))
-            self._cached_actor.stop.remote()
+            ray.get(self._cached_actor.stop.remote(), DEFAULT_GET_TIMEOUT)
```

Comment:
I think the problem with doing ray.get() is that it stalls the entire Tune event loop. Is it possible to achieve this without a blocking call?

Reply:
I see. I did not realize that. Probably there is but I am too poor in Ray to make more than such simple changes.

Reply:
So maybe I should close this pull request and just fill an issue for somebody more knowledgeable?

Reply:
Hm, the terminate call should only execute on the actor after “stop” is finished executing. Do you see errors?

Reply:
I don't see errors. I see that my `Trainable.stop`, which takes some time is interrupted during the execution because the actor is terminated without waiting the stop to finish. So, I guess, we need to wait for stop to finish before terminating the actor.

Reply:
This is odd.... each method should be invoked serially (i.e., one method will invoke at a time). Can you provide a reproducible script?